### ISSUE TITLE
az container exec. pipe command result to file fails [Errno 25] Inappropriate ioctl for device

### ISSUE THREAD

**Issue by programista-zacny** [Azure/azure-cli#28553]
### Describe the bug

I'm trying to
`az containerapp exec --subscription subscription -g group -n name --command "cat /etc/hosts" >> test.log` and it fails...
```
Exception in thread Thread-1 (read_ssh):
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/homebrew/Cellar/azure-cli/2.58.0/libexec/lib/python3.11/site-packages/azure/cli/command_modules/containerapp/_ssh_utils.py", line 123, in read_ssh
    _resize_terminal(connection)
  File "/opt/homebrew/Cellar/azure-cli/2.58.0/libexec/lib/python3.11/site-packages/azure/cli/command_modules/containerapp/_ssh_utils.py", line 154, in _resize_terminal
    size = os.get_terminal_size()
           ^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 25] Inappropriate ioctl for device
```

### Related command

az containerapp exec --subscription subscription -g group -n name --command "cat /etc/hosts" >> test.log

### Errors

Exception in thread Thread-1 (read_ssh):
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.8/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/homebrew/Cellar/azure-cli/2.58.0/libexec/lib/python3.11/site-packages/azure/cli/command_modules/containerapp/_ssh_utils.py", line 123, in read_ssh
    _resize_terminal(connection)
  File "/opt/homebrew/Cellar/azure-cli/2.58.0/libexec/lib/python3.11/site-packages/azure/cli/command_modules/containerapp/_ssh_utils.py", line 154, in _resize_terminal
    size = os.get_terminal_size()
           ^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 25] Inappropriate ioctl for device

### Issue script & Debug output

.

### Expected behavior

Pipe result to file.

### Environment Summary

{
  "azure-cli": "2.58.0",
  "azure-cli-core": "2.58.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {}
}

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by Greedygre** [comment]
Hi @programista-zacny 

Thanks for reporting this issue.
May I ask what is the device and scenario for executing the command? 
`OSError: [Errno 25] Inappropriate ioctl for device`

Thanks.

**Comment by programista-zacny** [comment]
MacOS Sonoma 14 on MacBook Air M2.

**Comment by Greedygre** [comment]
> MacOS Sonoma 14 on MacBook Air M2.
Hi @programista-zacny 

Can you help me to double confirm the result of the execution of these python code?
```
from shutil import get_terminal_size
size = get_terminal_size()
print(size) 
```

Thanks.

**Comment by programista-zacny** [comment]
Result: `os.terminal_size(columns=200, lines=50)`
I run this on iTerm.

**Comment by Greedygre** [comment]
> Result: `os.terminal_size(columns=200, lines=50)` I run this on iTerm.


Thanks.
The fix has been merged and will be released in Azure CLI version: 2.60.0.
https://github.com/Azure/azure-cli/milestone/141
